lib/battle/CBattleInfoCallback.cpp

```diff
@@ -526,7 +526,7 @@ bool CBattleInfoCallback::battleCanShoot(const CStack * stack, BattleHex dest) c
 			return false;
 	}
 
-	if(stack->getCreature()->idNumber == CreatureID::CATAPULT && dst) //catapult cannot attack creatures
+	if(stack->getCreature()->idNumber == CreatureID::CATAPULT) //catapult cannot attack creatures

```

Comment:
This code better to be moved above the:
```
	if(!stack || !dst)
		return false;
```

Reply:
But if we move this `if` above pointed check we may dereference invalid `stack`?

Reply:
Oops excuse me, I mean "below". It's just make more sense to check for the forgetfulness after creature type.